plugins/httpapi/mso.py

```diff
@@ -66,41 +77,42 @@ def login(self, username, password):
         # Perform login request
         self.connection.queue_message("vvvv", "Starting Login to {0}".format(self.connection.get_option("host")))
 
-        method = "POST"
-        path = "/mso/api/v1/auth/login"
-        full_path = self.connection.get_option("host") + path
-
-        if (self.params.get("login_domain") is not None) and (self.params.get("login_domain") != "Local"):
-            domain_id = self._get_login_domain_id(self.params.get("login_domain"))
-            payload = {"username": self.connection.get_option("remote_user"), "password": self.connection.get_option("password"), "domainId": domain_id}
-        else:
+        if self.connection._auth is None:
```

Comment:
That was my initial thought also, but thought I might be missing something since it is there in ansible-nd plugin: https://github.com/CiscoDevNet/ansible-nd/blob/master/plugins/httpapi/nd.py#L92.